### ISSUE TITLE
Graphql Aliases

### ISSUE THREAD

**Issue by Oupsla** [marmelab/react-admin#3160]
Hello folks 👋 
I'm using react-admin with ra-data-graphql-simple as dataProvider and I'm having an issue.
When I try to use aliases ([ref](https://graphql.org/learn/queries/#aliases)), the page fails to load.
I tried to find where it comes from but without success 😞 

**What you were expecting:**
Using aliases is working great and can be use like a normal query (the field is mapped to the new name defined by the alias)

Example (for a custom query defined for `fetchType: GET_ONE`)

```
query Device($id: ID!) {
  data: Device(id: $id) {
    id
    serial
    lat: latitude
    lng: longitude
    __typename
  }
}
```

**What happened instead:**
Got a `Warning: Missing translation for key: "Cannot read property 'type' of undefined"` in the console and the page fails to load and goes back to the list view.
But the query (over http) succeeds and returns correct result !

**Steps to reproduce:**
Create a dataprovider with ra-data-graphql-simple and add an alias.
Code sandbox : https://codesandbox.io/s/pkm8051kp0

Go to the customers list and see the error

**Environment**

* React-admin version: ^2.8.6
* Last version that did not exhibit the issue (if applicable):
* React version: ^16.8.6
* Browser: Chrome on Macos
* Stack trace (in case of a JS error):

```
console.(anonymous function) | @ | index.js:1446
-- | -- | --
  | printWarning | @ | warning.js:34
  | warning | @ | warning.js:58
  | warn | @ | index.js:28
  | push../node_modules/node-polyglot/index.js.Polyglot.t | @ | index.js:395
  | Notification.render | @ | Notification.js:151
  | finishClassComponent | @ | react-dom.development.js:8747
  | updateClassComponent | @ | react-dom.development.js:8715
  | beginWork | @ | react-dom.development.js:9364
  | performUnitOfWork | @ | react-dom.development.js:12372
  | workLoop | @ | react-dom.development.js:12402
  | renderRoot | @ | react-dom.development.js:12431
  | performWorkOnRoot | @ | react-dom.development.js:13028
  | performWork | @ | react-dom.development.js:12949
  | performSyncWork | @ | react-dom.development.js:12927
  | requestWork | @ | react-dom.development.js:12817
  | scheduleWorkImpl | @ | react-dom.development.js:12691
  | scheduleWork | @ | react-dom.development.js:12650
  | enqueueSetState | @ | react-dom.development.js:6920
  | push../node_modules/react/cjs/react.development.js.Component.setState | @ | react.development.js:336
  | onStateChange | @ | connectAdvanced.js:245
  | notify | @ | Subscription.js:29
  | notifyNestedSubs | @ | Subscription.js:66
  | onStateChange | @ | connectAdvanced.js:242
  | notify | @ | Subscription.js:29
  | notifyNestedSubs | @ | Subscription.js:66
  | onStateChange | @ | connectAdvanced.js:242
  | dispatch | @ | createStore.js:175
  | e | @ | VM273:1
  | (anonymous) | @ | middleware.js:13
  | (anonymous) | @ | formMiddleware.js:22
  | (anonymous) | @ | middleware.js:72
  | dispatch | @ | applyMiddleware.js:46
  | (anonymous) | @ | utils.js:282
  | (anonymous) | @ | proc.js:557
  | exec | @ | scheduler.js:20
  | flush | @ | scheduler.js:63
  | asap | @ | scheduler.js:35
  | runPutEffect | @ | proc.js:553
  | runEffect | @ | proc.js:496
  | next | @ | proc.js:366
  | currCb | @ | proc.js:449
  | Promise.then (async) |   |  
  | resolvePromise | @ | proc.js:515
  | runCallEffect | @ | proc.js:584
  | runEffect | @ | proc.js:496
  | next | @ | proc.js:366
  | currCb | @ | proc.js:449
  | checkEffectEnd | @ | proc.js:689
  | chCbAtKey | @ | proc.js:707
  | currCb | @ | proc.js:449
  | (anonymous) | @ | proc.js:567
  | exec | @ | scheduler.js:20
  | flush | @ | scheduler.js:63
  | asap | @ | scheduler.js:35
  | runPutEffect | @ | proc.js:553
  | runEffect | @ | proc.js:496
  | next | @ | proc.js:366
  | currCb | @ | proc.js:449
  | (anonymous) | @ | proc.js:516
  | Promise.then (async) |   |  
  | resolvePromise | @ | proc.js:515
  | runCallEffect | @ | proc.js:584
  | runEffect | @ | proc.js:496
  | next | @ | proc.js:366
  | currCb | @ | proc.js:449
  | checkEffectEnd | @ | proc.js:689
  | chCbAtKey | @ | proc.js:707
  | currCb | @ | proc.js:449
  | (anonymous) | @ | proc.js:567
  | exec | @ | scheduler.js:20
  | flush | @ | scheduler.js:63
  | asap | @ | scheduler.js:35
  | (anonymous) | @ | channel.js:215
  | emit | @ | channel.js:39
  | (anonymous) | @ | middleware.js:74
  | (anonymous) | @ | bindActionCreators.js:3
  | UnconnectedShowController.updateData | @ | ShowController.js:105
  | UnconnectedShowController.componentDidMount | @ | ShowController.js:87
  | commitLifeCycles | @ | react-dom.development.js:10251
  | commitAllLifeCycles | @ | react-dom.development.js:11994
  | callCallback | @ | react-dom.development.js:107
  | invokeGuardedCallbackDev | @ | react-dom.development.js:144
  | invokeGuardedCallback | @ | react-dom.development.js:191
  | commitRoot | @ | react-dom.development.js:12138
  | completeRoot | @ | react-dom.development.js:13086
  | performWorkOnRoot | @ | react-dom.development.js:13032
  | performWork | @ | react-dom.development.js:12949
  | performSyncWork | @ | react-dom.development.js:12927
  | interactiveUpdates | @ | react-dom.development.js:13193
  | interactiveUpdates | @ | react-dom.development.js:2057
  | dispatchInteractiveEvent | @ | react-dom.development.js:4474
```

**Comment by fzaninotto** [comment]
Aliases are not supported at this time. Feel free to submit a PR to add support for it.

**Comment by Oupsla** [comment]
Would be glad to help !
Could you point me where it should be done ?

**Comment by djhi** [comment]
Fixed by #3168
